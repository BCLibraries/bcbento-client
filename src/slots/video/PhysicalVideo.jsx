import React from 'react';
import FixBrokenImage from "../../FixBrokenImage";
import {PrimoRecordLink} from "../PrimoRecordLink";
import AvailabilityContainer from "../availability/AvailabilityContainer";

const defaultVHSIcon = 'https://library.bc.edu/images/video-tape.svg';
const defaultDVDIcon = 'https://library.bc.edu/images/mono-dvd-mount.svg';

const vhsClass = 'physical-video__vhs-cover',
    dvdClass = 'physical-video__dvd-cover';

let defaultIcon = defaultVHSIcon;
let iconClass = vhsClass;

function PhysicalVideo({video}) {
    if (video.format.includes('disc')) {
        defaultIcon = defaultDVDIcon;
        iconClass = dvdClass;
    }

    const cover = getCover(video);

    const altText = `Catalog record for ${video.title}`;

    const recordLink = PrimoRecordLink(video);

    return <div className="physical-video">
        <a href={recordLink} aria-hidden="true">
            <img src={cover}
                 onLoad={onImageLoad}
                 onError={FixBrokenImage(defaultIcon, `OPER physical-video__default_cover ${iconClass}`)}
                 alt={altText}
                 className={coverClass(video)}/>
        </a>

        <h3 className="physical-video__media-heading media-heading">
            <a href={recordLink} className="physical-video__title">
                {video.title}
            </a>
        </h3>

        {video.date}

        <div className="physical-video__creator">{creatorName(video)}</div>

        {video.format}

        <AvailabilityContainer item={video}/>

    </div>
}

function coverClass(video) {
    return (video.coverImages.length > 0 && video.coverImages[0]) ? 'physical-video__box_cover' : `OPER HERE NOW physical-video__default_cover  ${iconClass}`;
}

function getCover(video) {
    if (video.coverImages.length > 0 && video.coverImages[0]) {
        return String(video.coverImages[0].url);
    }

    return defaultIcon;
}

function creatorName(video) {
    if (video.creator) {
        return video.creator;
    }

    if (video.contributors[0]) {
        return video.contributors[0];
    }

    return '';
}

function onImageLoad(event) {
    if (event.target.naturalHeight === 1) {
        event.target.className = `physical-video__default_cover ${iconClass}`;
        event.target.src = defaultIcon;
    }
}

export default PhysicalVideo;