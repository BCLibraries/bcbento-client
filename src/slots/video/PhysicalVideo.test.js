import renderer from 'react-test-renderer';
import PhysicalVideo from './PhysicalVideo.jsx';

it('renders correctly', () => {
    const video = {
        "id": "ALMA-BC21386241170001021",
        "title": "Casablanca",
        "date": "1999",
        "type": "Video",
        "creator": null,
        "contributors": [
            "Michael Curtiz 1886-1962",
            "Julius J. Epstein 1909-2000",
            "Philip G Epstein",
            "Howard Koch",
            "Hal B. Wallis 1899-",
            "Humphrey Bogart 1899-1957",
            "Ingrid Bergman 1915-",
            "Paul Henreid",
            "Claude Rains 1889-1967",
            "Conrad Veidt 1893-1943",
            "Sydney Greenstreet",
            "Peter Lorre",
            "Arthur Edeson 1891-1970",
            "Owen Marks",
            "Max Steiner 1888-1971",
            "Murray Burnett",
            "Warner Bros. Pictures (1923-1967)",
            "Turner Entertainment Co",
            "Warner Home Video (Firm)"
        ],
        "coverImages": [
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=0790743132&issn=/MC.JPG&client=primo",
                "__typename": "Link"
            },
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=0790743132&issn=/MC.JPG&client=primo",
                "__typename": "Link"
            },
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=0790743132&issn=/MC.JPG&client=primo",
                "__typename": "Link"
            }
        ],
        "available": true,
        "isPhysical": true,
        "isElectronic": false,
        "linkableId": "ALMA-BC21386241170001021",
        "screenCap": null,
        "format": "1 videodisc (103 min.) : sd., b&w ; 4 3/4 in.",
        "mms": "99125653740001021",
        "availability": {
            "libraryName": "O'Neill",
            "locationName": "3rd Floor Course Reserves -3 Hour Loan",
            "totalCount": 1,
            "callNumber": "PN1997 .C38 1999",
            "otherAvailabilities": false,
            "__typename": "Availability"
        },
        "__typename": "CatalogItem"
    };

    const component = renderer.create(
        <PhysicalVideo video={video}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
