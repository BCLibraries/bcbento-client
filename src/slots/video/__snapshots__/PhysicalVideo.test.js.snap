// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="physical-video"
>
  <a
    aria-hidden="true"
    href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC21386241170001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
    tabIndex={-1}
  >
    <img
      alt="Catalog record for Casablanca"
      className="physical-video__box_cover"
      onError={[Function]}
      onLoad={[Function]}
      src="https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=0790743132&issn=/MC.JPG&client=primo"
    />
  </a>
  <div>
    <div
      className="catalog-result-item__type"
    >
      Video
    </div>
    <h3
      className="physical-video__media-heading media-heading"
    >
      <a
        className="physical-video__title"
        href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC21386241170001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
      >
        Casablanca
      </a>
    </h3>
  </div>
  1999
  <div
    className="physical-video__creator"
  >
    Michael Curtiz 1886-1962
  </div>
  1 videodisc (103 min.) : sd., b&w ; 4 3/4 in.
  <ul
    className="available-items-list"
  >
    <li
      className="item-info item-info--available"
    >
      <a>
        <span
          className="item-info__library"
        >
          Find in 
          O'Neill
        </span>
        <span
          className="item-info__location"
        >
          3rd Floor Course Reserves -3 Hour Loan
        </span>
        <span
          className="item-info__callno"
        >
          (
          PN1997 .C38 1999
          )
        </span>
        <span
          className="item-info__other-libraries"
        />
      </a>
    </li>
  </ul>
</div>
`;
